table/get_context.cc

```diff
@@ -273,19 +280,23 @@ bool GetContext::SaveValue(const ParsedInternalKey& parsed_key,
           }
         } else if (kMerge == state_) {
           assert(merge_operator_ != nullptr);
-          state_ = kFound;
-          if (do_merge_) {
-            if (LIKELY(pinnable_val_ != nullptr)) {
-              Status merge_status = MergeHelper::TimedFullMerge(
-                  merge_operator_, user_key_, &value,
-                  merge_context_->GetOperands(), pinnable_val_->GetSelf(),
-                  logger_, statistics_, clock_);
-              pinnable_val_->PinSelf();
-              if (!merge_status.ok()) {
-                state_ = kCorrupt;
+          if (is_blob_index_ != nullptr && *is_blob_index_) {
+            Slice blob_value = value;
+            Status status = get_blob_callback_(value, blob_value);
+            if (!status.ok()) {
+              if (status.IsIncomplete()) {
+                MarkKeyMayExist();
+                return false;
               }
+              state_ = kCorrupt;
+              return false;
             }
+            Merge(&blob_value);
+            *is_blob_index_ = false;
           } else {
+            Merge(&value);
+          }
+          if (do_merge_ == false) {
```

Comment:
If we're doing `GetMergeOperands` and it's a blob index, we would want to save the blob we retrieved as an operand, right?

Reply:
Oh, I forgot about ```DB::GetMergeOperands``` which covers ```do_merge = false```. I will add the test and fix the code.